### PR TITLE
Handle x86_64 os.arch for native libraries

### PR DESCRIPTION
**Pull request by rjernst**
On some systems Java appears to return amd64 (even if not an amd processor), but on others it returns x86_64. This commit handles the latter case to correctly associate the arch with the appropriate platform dir.

### CODE REVIEW

**Review by jpountz — APPROVED** [review]
This looks good to me. I wonder if we're missing other ones now, not sure how to check this.

**Review by ldematte — APPROVED** [review]
LGTM
Self reminder: we'll need something similar when we integrate x64 to native vec (after #106133)

**Comment by rjernst** [comment]
>  we'll need something similar when we integrate x64 to native vec

I don't think that is true. The vec code relies on NativeAccess to load the vec library. While vec uses multi release jars, it should not need to determine anything about the architecture, the return value from NativeAccess determines whether vec can be supported. As far as actually loading the library within NativeAccess, once x64 is supported, the conditionals on loading based on arch can be removed since both aarch64 and x64 will be supported.

**Comment by ldematte** [comment]
>I don't think that is true. The vec code relies on NativeAccess to load the vec library. While vec uses multi release jars, it should not need to determine anything about the architecture, the return value from NativeAccess determines whether vec can be supported. As far as actually loading the library within NativeAccess, once x64 is supported, the conditionals on loading based on arch can be removed since both aarch64 and x64 will be supported.

That's correct; we will need something similar, but in the gradle script(s) that put the generated libraries in the correct places/upload the artifacts, not in the actual Java code. ~~I'm not sure about removing the conditionals at load time: what if someone runs ES in some "fancy" environment~~ scratch that, there is no Solaris or RISC-V java download, so I guess it's only ARM and x64.

**Review by ChrisHegarty — APPROVED** [review]

File: distribution/tools/server-cli/src/main/java/org/elasticsearch/server/cli/SystemJvmOptions.java [hunk 1]
```diff
@@ -155,7 +155,7 @@ private static String findLibraryPath(Map<String, String> sysprops) {
         }
         String archname = sysprops.get("os.arch");
         String arch;
-        if (archname.equals("amd64")) {
+        if (archname.equals("amd64") || archname.equals("x86_64")) {
```
Comment (ChrisHegarty) [inline distribution/tools/server-cli/src/main/java/org/elasticsearch/server/cli/SystemJvmOptions.java:158]: I think that this should be fine, and cover the archs we run on. No need for archs scuh as, say, "i386", "x86", "ppc64", "ppc64le", "aarch64", "riscv64".

**Pull request merged**